File: Python/Object_example.py
class BankAccount:
    bank_name = "Simple Bank"  # class attribute

    def __init__(self, owner: str, balance: float = 0.0) -> None:
        self.owner = owner
        self.balance = balance

    def deposit(self, amount: float) -> None:
        if amount <= 0:
            raise ValueError("Deposit amount must be greater than 0.")
        self.balance += amount

    def withdraw(self, amount: float) -> None:
        if amount <= 0:
            raise ValueError("Withdraw amount must be greater than 0.")
        if amount > self.balance:
            raise ValueError("Insufficient balance.")
        self.balance -= amount

    def get_balance(self) -> float:
        return self.balance

    @classmethod
    def from_string(cls, data: str) -> "BankAccount":
        # Expected format: "owner,balance"
        owner, balance = data.split(",")
        return cls(owner.strip(), float(balance.strip()))

    @staticmethod
    def is_valid_amount(amount: float) -> bool:
        return amount > 0

    def __str__(self) -> str:
        return f"Account(owner={self.owner}, balance={self.balance:.2f})"


def main() -> None:
    # __init__ + instance methods
    account = BankAccount("Daniel", 100.0)
    account.deposit(50.0)
    account.withdraw(30.0)
    print(account)  # __str__

    # class method
    second_account = BankAccount.from_string("Alice,250")
    print(second_account)

    # static method
    print("Valid amount 10?", BankAccount.is_valid_amount(10))
    print("Valid amount -5?", BankAccount.is_valid_amount(-5))

    # regular getter method
    print("Current balance:", account.get_balance())


if __name__ == "__main__":
    main()

File: Typescript/Object_example.ts
class BankAccount {
  static bankName = 'Simple Bank';

  constructor(owner, balance = 0) {
    this.owner = owner;
    this.balance = balance;
  }

  deposit(amount) {
    if (amount <= 0) {
      throw new Error('Deposit amount must be greater than 0.');
    }
    this.balance += amount;
  }

  withdraw(amount) {
    if (amount <= 0) {
      throw new Error('Withdraw amount must be greater than 0.');
    }
    if (amount > this.balance) {
      throw new Error('Insufficient balance.');
    }
    this.balance -= amount;
  }

  getBalance() {
    return this.balance;
  }

  static fromString(data) {
    const [owner, balance] = data.split(',');
    return new BankAccount(owner.trim(), Number(balance.trim()));
  }

  static isValidAmount(amount) {
    return amount > 0;
  }

  toString() {
    return `Account(owner=${this.owner}, balance=${this.balance.toFixed(2)})`;
  }
}

const main = () => {
  const account = new BankAccount('Daniel', 100);
  account.deposit(50);
  account.withdraw(30);
  console.log(account.toString());

  const secondAccount = BankAccount.fromString('Alice,250');
  console.log(secondAccount.toString());

  console.log('Valid amount 10?', BankAccount.isValidAmount(10));
  console.log('Valid amount -5?', BankAccount.isValidAmount(-5));

  console.log('Current balance:', account.getBalance());
};

main();
